include/mbedtls/mbedtls_config.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * \file mbedtls_config.h
+ *
+ * \brief Configuration options (includable header)
+ *
+ *  Include this file if you need to depend on the
+ *  configuration options defined in mbedtls_config_defines.h
+ */
+ /*
+  *  Copyright The Mbed TLS Contributors
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  */
+
+#ifndef MBEDTLS_MBEDTLS_CONFIG_H
+#define MBEDTLS_MBEDTLS_CONFIG_H
+
+#if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_DEPRECATE)
+#define _CRT_SECURE_NO_DEPRECATE 1
+#endif
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/mbedtls_config_defines.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if !defined(MBEDTLS_CONFIG_VERSION) || \
```

Comment:
I'm not sold on _requiring_ `MBEDTLS_CONFIG_VERSION` to be set. Why don't we allow config files that don't set it, for users who just want certain features to be enabled and don't care about details? It's not like we're planning to change `MBEDTLS_CONFIG_VERSION` often anyway.

In other words, I think that a config file with `MBEDTLS_CONFIG_VERSION` undefined should mean “I want this configuration, fill in any blanks in a sensible way”, while a config file with `MBEDTLS_CONFIG_VERSION` set should mean “I want this configuration exactly and I'm very sensitive about backward compatibility and not getting additional stuff that could increase code size”.